### ISSUE TITLE
v3: Filter Plugin Search Input Counted as Input in Multiselect

### ISSUE THREAD

**Issue by tdouglas1313** [ehynds/jquery-ui-multiselect-widget#849]
### What is the current behavior?

The multiselect includes the filter search input when getting inputs. such has get unChecked or anything else related to inputs.  

### What is the expected behavior?

it would not include the search input box when handling inputs

### If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo

create a multiselect of n options with filtering enabled. then getUnchecked.length + getChecked.length equals n+1 when it should be n

and most of the time the value of the filter text field is in the unchecked results

this also causes issues for things like # of # for the multiselect label.  

### Please tell us about your setup
  
  * Version: 3.0.0
  * Browser: Chrome

**Comment by mlh758** [comment]
I opened a PR that should fix your issue. Mind pulling it down and verifying that this works for you?

**Comment by tdouglas1313** [comment]
so that did not fix getUnchecked and getChecked. I thought about adding something like 
```javascript
return this.$inputs.filter("[type!=\"search\"]:checked");
```
to those methods, but I don't like how specific that is to removing the search field. it is possible that you could also do something like 
```javascript
return this.$inputs.filter("[type=\"checkbox\"]:checked,[type=\"radio\"]:checked");
``` 
though I haven't testing the second one. that also has issues with having to do that everywhere an input is queried for using jQuery. 

 Unfortunately the issue isn't isolated to `getUnchecked` and `getChecked`. it is present in other places like `selectedText` both strings that contain `# of #` and as a `function`

**Comment by mlh758** [comment]
Are you sure you pulled my branch down? I verified it with `getChecked` and `getUnchecked`, and have unit tests added to it as well that I verified failed before my changes and passed after.

**Comment by mlh758** [comment]
Also, `this.$inputs` already excludes the search input.

Using the filter demo for example:
```javascript
var test = $($('select')[0])
var inst = test.multiselect("instance")
inst.$inputs
```

gives you 8 inputs and they are all options from the select.

**Comment by mlh758** [comment]
Ah but apparently if you call `refresh` it adds the input back in.

**Comment by tdouglas1313** [comment]
I do call refresh on the multiselects I have been looking at. so maybe that is the reason this seems to always show up for me.

**Comment by mlh758** [comment]
I pushed another commit up with another test case added. It should continue to remain excluded after a refresh.

I added a css class to the search input's label and the this.$labels collection is now set as `this.$labels = this.$menu.find('label:not(.search-label)');` to keep that one from dropping in. Most of the internal methods look at those caches for the inputs.

I'm going to change that class name to be more specific to this widget and avoid name conflicts though.

**Comment by tdouglas1313** [comment]
I just pulled and tested what I have been looking at it seems to have fixed it. thanks for your help on this.

**Comment by mlh758** [comment]
Cool, I'll merge and tag it now.

**Comment by mlh758** [comment]
I had been meaning to do the real release of v3 anyway so this is available on npm as v3.0.0
